pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java

```diff
@@ -65,13 +66,14 @@ public class ColumnMetadataImpl implements ColumnMetadata {
   private final int _totalNumberOfEntries;
   private final PartitionFunction _partitionFunction;
   private final Set<Integer> _partitions;
+  private final Map<ColumnIndexType, Long> _indexSizeMap;
```

Comment:
adding this map here should allow us to report index info (only size for now)

Reply:
planning to move IndexKey or IndexEntry (without the buffer) from segment-local to segment-spi. thoughts?

Reply:
resolved. no need for this